splash/browser_tab.py

```diff
@@ -287,6 +289,11 @@ def go(self, url, callback, errback, baseurl=None, http_method='GET',
         """
         self.store_har_timing("_onStarted")
 
+        if body and http_method not in ["POST", "PUT"]:
+            # if someone gives body but forgets method
+            # QT will raise exceptions
+            http_method = "POST"
```

Comment:
Wouldn't it be better to throw the exception if the programmer made a mistake, instead of guessing?  Also it looks like it's doing that below with `ValueError("Request with method {} should not have body".format(method))`